### ISSUE TITLE
Converting a MOS object with a circular structure

### ISSUE THREAD

**Issue by PascalViauRC** [nrkno/sofie-mos-connection#83]
*  I'm submitting a ...
[x] bug report
[ ] feature request
[ ] question about the decisions made in the repository
[ ] question about how to use this project

I'm doing some tests with an iNews Mos Gateway. When there is no MOS object in the stories, everything works fine. When I add a MOS object to a story, I get this error while it converts XML to JSON  :

<code>TypeError: Converting circular structure to JSON\n    --> starting at object with constructor 'Object'\n    |     property 'mos' -> object with constructor 'Object'\n    --- property 'parent' closes the circle</code>

The problem seems to be that the iNews MOS object starts with the same ```<mos>``` root tag than the iNews story itself :

```xml
<mos>
  ...
    <roStorySend>
    ...
      <mosExternalMetadata>
        <mosPayload>
        ...
          <nsml xmlns:nsml="http://www.avid.com">
          ...
            <aeset-atts>
              <attachment id="1">
                <AttachmentContent>
                  <mos>
                    <mosID>
		         VIZRT
	            </mosID>
                     ...
                  </mos>
                </AttachmentContent>
              </attachment>
            </aeset-atts>
         </nsml>
      </mosPayload>
    </mosExternalMetadata>
  </roStorySend>
</mos>
```

When I replace the MOS object first tag (```<mos>```) by a modified tag (```<_mos>```), it fixes the problem : 

```xml
<mos>
  ...
    <roStorySend>
    ...
      <mosExternalMetadata>
        <mosPayload>
        ...
          <nsml xmlns:nsml="http://www.avid.com">
          ...
            <aeset-atts>
              <attachment id="1">
                <AttachmentContent>
                  <_mos>
                    <mosID>
		         VIZRT
	            </mosID>
                     ...
                  </_mos>
                </AttachmentContent>
              </attachment>
            </aeset-atts>
         </nsml>
      </mosPayload>
    </mosExternalMetadata>
  </roStorySend>
</mos>
```

If you want me to submit a pull request with the code I used to fix the problem, let me know !

**Comment by nytamin** [comment]
hmm, that's a strange bug indeed!

I did a little digging and it seems like there is a bug in the code that deals with reading the incoming message chunks. I think I've found the root cause and have implemented a fix in the PR #85.

@PascalViauRC would you be able to give that a spin (`npm install mos-connection@3.0.5-nightly-fix-multiple-mos-tags-20231214-120351-59cacb2.0`), to see if it helps?

**Comment by PascalViauRC** [comment]
@nytamin It fixes the problem, thanks !
